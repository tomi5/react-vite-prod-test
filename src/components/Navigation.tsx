import React from 'react';
import NavItem from './NavItem';

const Navigation: React.FC = () => {
  return (
    <nav className="bg-white shadow-md">
      <div className="max-w-6xl mx-auto px-4">
        <ul className="flex space-x-8 py-4">
          <NavItem to="/">Home</NavItem>
          <NavItem to="/about">About</NavItem>
          <NavItem to="/articles">Artyku≈Çy</NavItem>
        </ul>
      </div>
    </nav>
  );
};

export default Navigation;
