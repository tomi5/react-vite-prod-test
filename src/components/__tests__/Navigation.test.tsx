import { describe, it, expect, beforeEach } from 'vitest';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Navigation from '../Navigation';

describe('Navigation', () => {
  let container: HTMLElement;

  beforeEach(() => {
    const renderResult = render(
      <BrowserRouter>
        <Navigation />
      </BrowserRouter>,
    );
    container = renderResult.container;
  });

  it('should render all navigation links', () => {
    expect(screen.getByText('Home')).toBeInTheDocument();
    expect(screen.getByText('Artykuły')).toBeInTheDocument();
    expect(screen.getByText('About')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should have correct links', () => {
    expect(screen.getByText('Home').closest('a')).toHaveAttribute('href', '/');
    expect(screen.getByText('Artykuły').closest('a')).toHaveAttribute('href', '/articles');
    expect(screen.getByText('About').closest('a')).toHaveAttribute('href', '/about');
    expect(container).toMatchSnapshot();
  });
});
