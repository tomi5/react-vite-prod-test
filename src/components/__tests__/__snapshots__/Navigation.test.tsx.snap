// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navigation > should have correct links 1`] = `
<div>
  <nav
    class="bg-white shadow-md"
  >
    <div
      class="max-w-6xl mx-auto px-4"
    >
      <ul
        class="flex space-x-8 py-4"
      >
        <li>
          <a
            class="text-gray-600 hover:text-blue-600 font-medium"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            class="text-gray-600 hover:text-blue-600 font-medium"
            href="/about"
          >
            About
          </a>
        </li>
        <li>
          <a
            class="text-gray-600 hover:text-blue-600 font-medium"
            href="/articles"
          >
            Artykuły
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;

exports[`Navigation > should render all navigation links 1`] = `
<div>
  <nav
    class="bg-white shadow-md"
  >
    <div
      class="max-w-6xl mx-auto px-4"
    >
      <ul
        class="flex space-x-8 py-4"
      >
        <li>
          <a
            class="text-gray-600 hover:text-blue-600 font-medium"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            class="text-gray-600 hover:text-blue-600 font-medium"
            href="/about"
          >
            About
          </a>
        </li>
        <li>
          <a
            class="text-gray-600 hover:text-blue-600 font-medium"
            href="/articles"
          >
            Artykuły
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
